scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeDBConfiguration.java

```diff
@@ -0,0 +1,796 @@
+                                                                                  "notification",
+                                                                                  "notification");
+
+    private static final String VALID_HOSTNAME_REGEXP = "^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\\-]*[a-zA-Z0-9])\\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\\-]*[A-Za-z0-9])$";
+
+    private static final String VALID_IP_REGEXP = "^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])$";
+
+    private static final Pattern hostnamePattern = Pattern.compile(VALID_HOSTNAME_REGEXP);
+
+    private static final Pattern ipPattern = Pattern.compile(VALID_IP_REGEXP);
+
+    private static boolean isDebug = false;
+
+    private static String component = null;
+
+    private static String vendor = null;
+
+    private static String schemaName = null;
+
+    private static String username = null;
+
+    private static String password = null;
+
+    private static String url = null;
+
+    private static String hostname = null;
+
+    private static String port = null;
+
+    private static String timezone = null;
+
+    /**
+     * Entry point
+     *
+     * @param args arguments, try '-h' for help
+     * @see WebProperties
+     */
+    public static void main(String[] args) {
+
+        configurationFilesPerTarget.put("rm", "config/rm/database.properties");
+        configurationFilesPerTarget.put("scheduler", "config/scheduler/database.properties");
+        configurationFilesPerTarget.put("catalog", "dist/war/catalog/WEB-INF/classes/application.properties");
+        configurationFilesPerTarget.put("service-automation",
+                                        "dist/war/cloud-automation-service/WEB-INF/classes/application.properties");
+        configurationFilesPerTarget.put("job-planner", "dist/war/job-planner/WEB-INF/classes/application.properties");
+        configurationFilesPerTarget.put("notification",
+                                        "dist/war/notification-service/WEB-INF/classes/application.properties");
+        configurationFilesPerTarget.put("proactive-cloud-watch",
+                                        "dist/war/proactive-cloud-watch/WEB-INF/classes/application.properties");
+        configurationFilesPerTarget.put("scheduling-api",
+                                        "dist/war/scheduling-api/WEB-INF/classes/application.properties");
+
+        try {
+            if (changeDBConfiguration(args)) {
+                log(vendor + " database configuration successfully applied to " + component + ".");
+            }
+        } catch (ChangeDBConfigurationException e) {
+            System.err.println("ERROR : " + e.getMessage());
+            if (e.getCause() != null) {
+                System.err.println(e.getCause().getMessage());
+            }
+            if (e.getAdditionalInfo() != null) {
+                log(e.getAdditionalInfo());
+            }
```

Comment:
I suggest to display such error details only in debug mode.

Reply:
I don't see why, even not in debug mode, I want to see the full error details

Reply:
Both cases are correct. In many systems, in the debug/verbose mode we have more detailed outputs or errors